### ISSUE TITLE
Invalid link to GitHub on coala API Documentation

### ISSUE THREAD

**Issue by li-boxuan** [coala/coala#5316]
coala uses ```Read the Docs``` to generate API webpages. However, there is a floating window at the bottom right corner which displays a link to corresponding GitHub page.

For example, on the page [coalib package](http://api.coala.io/en/latest/coalib.html), you can click on the small floating window at the bottom right corner and it would display a window as follows:
<img src="https://user-images.githubusercontent.com/25746010/38401473-f301ce78-3923-11e8-9418-8e12c9d96638.png" alt="floating window" width="250">

The ```view``` link points to https://github.com/coala/coala/blob/master/docs/coalib.rst, which doesn't exist.

**Comment by jayvdb** [comment]
Is there a readthedocs config setting we can use to de-activate the 'On GitHub' section for these generated files?

**Comment by li-boxuan** [comment]
@jayvdb Seems not. Relevant issue: https://github.com/sphinx-doc/sphinx/issues/2386

**Comment by jayvdb** [comment]
An issue closer to our needs: https://github.com/rtfd/readthedocs-sphinx-ext/issues/28

and it looks like @flying-sheep has a hack at https://gist.github.com/flying-sheep/b65875c0ce965fbdd1d9e5d0b9851ef1 which we should investigate.

**Comment by flying-sheep** [comment]
That one only works for the “Edit on GitHub” link in the top right corner. See it working at e.g. http://anndata.readthedocs.io/en/latest/anndata.read_csv.html#anndata.read_csv

It would be cool if you could figure out how to make it also apply to the popup links!

**Comment by chay2199** [comment]
@li-boxuan The issue resolves by changing http://api.coala.io/en/latest/_static/custom.css using inspect on my browser but I cannot actually change this file, is this why we have to find a readthedocs config setting?

**Comment by KVGarg** [comment]
@li-boxuan The GitHub link provided is generated using the [current URL path](https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/restapi/views/footer_views.py#L119). 
Three workarounds to this solution:
1. Either disable `On Github` section
2. Add a `.rst` file for each file or
3. Add an upstream method to check whether the file is there or not in the directory. If not, provide a link to GitHub repository to that particular directory.

**Comment by KVGarg** [comment]
> The issue resolves by changing http://api.coala.io/en/latest/_static/custom.css using inspect on my browser

Do you mean by changing `href` attribute of anchor tag!

**Comment by chay2199** [comment]
@KVGarg adding this in custom.css disables the On Github section `div.rst-versions.shift-up > div.rst-other-versions > dl:nth-child(4) {
display: none;
}`

**Comment by KVGarg** [comment]
> @KVGarg adding this in custom.css disables the On Github section `div.rst-versions.shift-up > div.rst-other-versions > dl:nth-child(4) { display: none; }`

Changing `css` file won’t be a good solution to this problem i guess. You should probably take a look at sphinx configuration file if it need some changes to disable this section.

**Comment by chay2199** [comment]
@KVGarg Yes I believe that too. But I wasn't able to find any direct solution to disable this section in the config file at that time.

**Comment by li-boxuan** [comment]
Is it possible to find out the upstream bug?

**Comment by KVGarg** [comment]
> Is it possible to find out the upstream bug?

I guess it's not a bug because we've been creating coalib `.rst` files for each package in [setup.py](https://github.com/coala/coala/blob/master/setup.py#L113-L140) due to which no warning is being raised. And due to presence of these files in the readthedocs environment the html page is being rendered.
We can possibly add upstream enhancements in either `alabaster` or `readthedocs`:
1. In `alabaster`, we can provide a enhancement PR that adds an option to disable the `On GitHUb` section in injected footer. Or,
2. In `readthedocs`, a PR that provides a new algorithm that checks whether the file is there or not in the directory. If not, provide a link to GitHub repository to that particular directory. If both of them isn't found, the section won't be displayed!
3. Or we can change our theme to `sphinx_rtd_theme` in which we can disable the `On GitHub` section
